src/ai/backend/manager/registry.py

```diff
@@ -872,10 +872,13 @@ async def enqueue_session(
                 # the first kernel_config is repliacted to sub-containers
                 assert kernel_enqueue_configs[0]["cluster_role"] == DEFAULT_ROLE
                 kernel_enqueue_configs[0]["cluster_idx"] = 1
+                kernel_enqueue_configs[0]["local_rank"] = 0
                 for i in range(cluster_size - 1):
+                    increased_index = i + 1
                     sub_kernel_config = cast(KernelEnqueueingConfig, {**kernel_enqueue_configs[0]})
                     sub_kernel_config["cluster_role"] = "sub"
-                    sub_kernel_config["cluster_idx"] = i + 1
+                    sub_kernel_config["cluster_idx"] = increased_index
+                    sub_kernel_config["local_rank"] = increased_index
```

Comment:
Let's set `local_rank` from 0 not 1 because MPI local rank usually count from 0

Reply:
Done, Thanks